aten/src/ATen/native/composite/soft_margin_loss.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+#include <ATen/core/Reduction.h>
+#include <ATen/core/TensorBody.h>
+
+namespace at {
+namespace native {
+
+template <typename OPS>
+Tensor& soft_margin_loss_out(
+    const Tensor& input,
+    const Tensor& target,
+    int64_t reduction,
+    Tensor& output) {
+  // compute inplace variant of: output = at::log(1. + at::exp(-input *
+  // target));
+  OPS::neg_out(output, input);
+  OPS::mul_(output, target);
+  OPS::exp_(output);
+  OPS::add_(output, 1.);
+  OPS::log_(output);
```

Comment:
Don't know if it's supported by this new dispatch-less kernels, but it'd be great to use `log1p` here for numerical stability.